drivers/crypto/crypto_nrf52_ecb.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <string.h>
+#include <crypto/cipher.h>
+#include <logging/log.h>
+#include <hal/nrf_ecb.h>
+
+#define AES_KEY_SIZE   16
+#define AES_BLOCK_SIZE 16
+
+#define ECB_PERIPHERAL             NRF_ECB
+
+#define ECBDATA_KEY_OFFSET         0
+#define ECBDATA_CLEARTEXT_OFFSET   16
+#define ECBDATA_CIPHERTEXT_OFFSET  32
+
+LOG_MODULE_REGISTER(crypto_nrf52_ecb, CONFIG_CRYPTO_LOG_LEVEL);
+
+struct nrf52_ecb_drv_state_t {
+	u8_t ecb_data[48];    ///< ECB data structure for RNG peripheral
+	u8_t *key;            ///< Key:        Starts at ecb_data
+	u8_t *cleartext;      ///< Cleartext:  Starts at ecb_data + 16 bytes.
+	u8_t *ciphertext;     ///< Ciphertext: Starts at ecb_data + 32 bytes.
+	int in_use;
+};
+
+static struct nrf52_ecb_drv_state_t nrf52_ecb_drv_state;
+
+static int do_ecb_encrypt(struct cipher_ctx *ctx, struct cipher_pkt *pkt);
```

Comment:
The forward declaration does not match implementation (missing static). Is it really needed though? From what I've seen you could simply move the implementation above.